### ISSUE TITLE
RFE: OpenSSL/LibreSSL as an alternative to NSPR/NSS or beecrypt

### ISSUE THREAD

**Issue by Redfoxmoon3** [rpm-software-management/rpm#119]
OpenSSL/LibreSSL is a lot easier to build than NSPR/NSS and beecrypt, asking as it would simplify building for new platforms like midipix.org and lesser known posix-y platforms

**Comment by Conan-Kudo** [comment]
I believe @sgallagher from the Fedora Modularity group also was interested in this... 

Snippet from `#rpm.org` conversation back in October requesting it:
```
[Monday, October 17, 2016] [12:54:05 PM EDT] <sgallagh> ffesti: I know this is going to be a controversial question, but how hard would it be to switch RPM from using Mozilla NSS to using OpenSSL instead?
[Monday, October 17, 2016] [12:54:42 PM EDT] <ffesti>   well, easy
[Monday, October 17, 2016] [12:54:43 PM EDT] <sgallagh> RPM is actually the only package in the current vision of the Base Runtime module in Fedora that pulls in NSS. It would be awesome if we could eliminate one of the crypto libraries.
[Monday, October 17, 2016] [12:54:58 PM EDT] <sgallagh> Oh?
[Monday, October 17, 2016] [12:55:01 PM EDT] <ffesti>   if someone can implement the backend in ssl
[Monday, October 17, 2016] [12:55:34 PM EDT] <sgallagh> ffesti: What would that entail?
[Monday, October 17, 2016] [12:56:01 PM EDT] <ffesti>   see rpmio/digest*
```

**Comment by sgallagher** [comment]
Thanks for the ping. Yes, as it happens I just started working on implementing this. I hope to have something to share within the next two weeks.

**Comment by pmatilai** [comment]
The problem is licensing.

**Comment by sgallagher** [comment]
The licensing situation is solvable: https://fedoraproject.org/wiki/Licensing:FAQ?rd=Licensing/FAQ#What.27s_the_deal_with_the_OpenSSL_license.3F

Short version: In Fedora, the legal team has come to the determination that the usage is within the spirit of the licenses, if not its specific letter. That said, RPM could also opt to add a license exception for OpenSSL, which is supported by the FSF.

**Comment by sgallagher** [comment]
I just opened PR #129 

Reviews and concerns are most welcome.

**Comment by Conan-Kudo** [comment]
@Redfoxmoon3 OpenSSL support has now been merged into rpm git master with commit 64028f9a1c25ada8ffc7a48775f526600edcbf85. 🎉

**Comment by robert-scheck** [comment]
> The licensing situation is solvable: https://fedoraproject.org/wiki/Licensing:FAQ?rd=Licensing/FAQ#What.27s_the_deal_with_the_OpenSSL_license.3F

What about the license exception mentioned at the URL above? Any chance to have that one for RPM?
